Fixes to pass my integration tests

Jarod42 opened the pull request:
Add support for:
- sysincludedirs, forceinclude
- pre/post build command
- customcommands

Fix link to sibbling projects

Tested with https://github.com/Jarod42/premake-sample-projects

Jarod42 committed:
- Handle `sysincludedirs` and `forceincludes`.
- Remove extra quotes for `set_target_properties` which were problematic when buildoptions already got quotes.

Jarod42 committed:
Clean output: don't show empty "property".

Jarod42 committed:
Use project name instead of linktarget in `target_link_libraries`, so library path is correct.

Jarod42 committed:
Add support for pre/post buildcommands.

Jarod42 committed:
Add support for custom commands.

Jarod42 committed:
Add support for rules.

Jarod42 committed:
Replace some absolute paths by relative paths.

Jarod42 committed:
Handle project without source files.

Jarod42 committed:
Fix case (might be problematic for override of premake module such as premake-qt)

Jarod42 committed:
Fix single/double quote in *`buildmessage`.

Jarod42 committed:
Handle "missing" directories for custom command.

Jarod42 committed:
Use `externalincludedirs` instead of deprecated `sysincludedirs`.

Jarod42 committed:
Handle linkoptions for `sanitize { "Address", "Fuzzer" }`.

Jarod42 committed:
cflags/cppflags handles both msvc and gcc toolset.

Jarod42 committed:
Fix single quote in messages (as premake-qt adds).

Jarod42 committed:
- Fix some multi-configuration issues.
- Fix indentation.
- Fix empty test for `cfg.sanitize`.

Jarod42 committed:
Add generated files only when `filecfg.compilebuildoutputs` is true.

Jarod42 committed:
Add generated file as dependencies of app.

Jarod42 committed:
Add support of `frameworkdirs` and `includedirsafter`

Jarod42 committed:
Add prelink support.

Jarod42 committed:
Handle `flags {"ExcludeFromBuild"}`.

Jarod42 committed:
Fix generated files in different directory depending of config.

Jarod42 committed:
Add support to `undefines`.

Jarod42 committed:
Default toolset depends on target OS and fix toolset with version

Jarod42 committed:
Handle multi-config generators

Jarod42 committed:
Handle invalid char in configuration/platform name (as hyphen) (`$<CONFIG:Release-x86>` produces errors)

Jarod42 committed:
Fix pre/post build commands by config.

Margen67 commented on _preload.lua:
```diff
@@ -20,14 +20,22 @@ p.api.register {
 }
 
+local default_toolset_map = {
+	["windows"] = "msc-v142", -- Visual Studio 2019
```
```suggestion
	["windows"] = "msc-v143", -- Visual Studio 2022
```

Jarod42 committed:
Replace removed `flags` by the dedicated API

alchemyyy committed:
Updated to support the full premake5 API surface

_preload.lua**: Added None and SharedItems to valid_kinds. Updated default MSVC toolset to v143 (VS2022). Updated action description.

**cmake.lua**: Extended generateProject to emit .cmake files for Utility, None, Makefile, and SharedItems project kinds (previously only C/C++).

**cmake_workspace.lua**: Implemented startproject support via VS_STARTUP_PROJECT (was a TODO).

**cmake_project.lua**: Major expansion. Added cross-platform command tokens using `${CMAKE_COMMAND} -E` instead of host-specific shell commands (fixes {COPYFILE}, {COPYFILEIFNEWER}, {MKDIR}, etc.). Fixed add_custom_command generation and pch relative paths. Added project kind handling for Utility/None (custom targets), SharedItems (interface libraries), Makefile (custom targets), WindowedApp (WIN32 flag). Made pic and lto independent of cppdialect. Added support for: cdialect, optimize, symbols, debugformat, warnings, externalwarnings, enablewarnings, disablewarnings, fatalwarnings, rtti, exceptionhandling, floatingpoint, visibility, inlinesvisibility, omitframepointer, unsignedchar, openmp, structmemberalign, buffersecuritycheck, strictaliasing, editandcontinue, multiprocessorcompile, vectorextensions, isaextensions, characterset, staticruntime/runtime (MSVC_RUNTIME_LIBRARY), compileas, targetprefix/targetsuffix/targetextension, objdir, dependson, syslibdirs, runpathdirs, entrypoint, linker (LLD), linkerfatalwarnings, systemversion, sanitize (expanded with Thread and UndefinedBehavior), and all C++23/gnu++23/intermediate dialect variants.

Jarod42 committed:
Merge pull request #6 from alchemyyy/master

Updated to support the "full" premake5 API surface

alex-rass-88 committed:
Fix rtti flag.

rtti flag should be used only in c++ code.

Jarod42 committed:
Merge pull request #7 from alex-rass-88/master

Fix rtti flag.

Jarod42 committed:
Fix dev-warning

Jarod42 committed:
Disable suffix part to let tests pass
